root/usr/lib64/ruby/vendor_ruby/discovery/screen/facts.rb

```diff
@@ -46,7 +41,7 @@ def screen_facts mac, proxy_url, proxy_type
     if perform_upload(proxy_url, proxy_type, custom_facts)
       [:screen_status, generate_info(' - ' + _('awaiting kexec into installer'))]
     else
-      :screen_welcome
+      [:screen_status, generate_info(' - ' + _('fact upload failed'))]
```

Comment:
This is also an improvement - when PXE-less upload fails, you are taken to status page which is better than welcome page (to investigate logs or resend and stuff).